packages/mdc-textfield/mdc-textfield.scss

```diff
@@ -540,72 +607,87 @@ $mdc-textfield-disabled-border-on-dark: rgba(white, .3);
       }
     }
   }
-  // stylelint-enable plugin/selector-bem-pattern
-}
 
-.mdc-textfield--fullwidth {
-  display: block;
-  box-sizing: border-box;
-  width: 100%;
-  margin: 0;
-  border: none;
-  border-bottom: 1px solid $mdc-textfield-divider-on-light;
-  outline: none;
+  &:not(.mdc-textfield--upgraded) {
+    .mdc-textfield__input {
+      padding-top: $padding-inset;
+    }
+  }
 
-  // stylelint-disable plugin/selector-bem-pattern
+  .mdc-textfield__icon {
+    @include mdc-rtl-reflexive-position(right, 22px);
 
-  &:not(.mdc-textfield--multiline) {
-    height: 56px;
+    position: absolute;
+    top: 18px;
+    color: $mdc-textfield-underline-on-light-idle;
+    cursor: pointer;
+
+    @include mdc-theme-dark(".mdc-textfield") {
+      color: $mdc-textfield-dark-label;
+    }
   }
+}
 
-  &.mdc-textfield--multiline {
-    padding: 20px 0 0;
+.mdc-textfield--textarea-with-icon {
+  .mdc-textfield__input {
+    padding-right: 54px;
   }
+}
 
-  &.mdc-textfield--dense {
-    &:not(.mdc-textfield--multiline) {
-      height: 48px;
-    }
+.mdc-textfield--textarea.mdc-textfield--focused {
+  @include mdc-theme-prop(border-color, primary);
+}
+
+.mdc-textfield--textarea-with-icon.mdc-textfield--focused {
+  .mdc-textfield__icon {
+    @include mdc-theme-prop(color, text-primary-on-light);
 
-    &.mdc-textfield--multiline {
-      padding: 16px 0 0;
+    @include mdc-theme-dark(".mdc-textfield") {
+      @include mdc-theme-prop(color, secondary);
```

Comment:
This seems like it's assuming that secondary is a light color. Would text-primary-on-dark make more sense?

Reply:
This is all out. See comment about about textarea.